starknet_py/proxy_check.py

```diff
@@ -0,0 +1,42 @@
+from abc import ABC, abstractmethod
+
+from starknet_py.net.models import parse_address
+
+
+class ProxyCheck(ABC):
+    @abstractmethod
+    async def is_proxy(self, contract: "Contract") -> bool:
+        """
+        :return: `True` if contract is a proxy or `False` if it is not
+        """
+
+    @abstractmethod
+    async def implementation_address(self, contract: "Contract") -> int:
+        """
+        :return: Address of contract being proxied by `contract` given as a parameter
+        """
+
+
+class ArgentProxyCheck(ProxyCheck):
+    async def is_proxy(self, contract: "Contract") -> bool:
+        return "get_implementation" in contract.functions
+
+    async def implementation_address(self, contract: "Contract") -> int:
+        res = await contract.functions["get_implementation"].call()
+        return res[0]
+
+
+class OpenZeppelinProxyCheck(ProxyCheck):
+    def __init__(self):
+        self.storage_key = (
+            1662968741416162122709795668327058413808168082301235317239158738933362048792
```

Comment:
how is this computed?

Reply:
https://starknet.io/docs/hello_starknet/cli.html#get-storage-at

Should I add this calculation to the constructor?

Reply:
yeaah probably. is this memory access something we want to ditch in the future? if that's correct please make a task to correct this and mark it here

Reply:
as FIXME

Reply:
So while this is a imperfect solution, it's uncertain if/when OZ adds a simpler way of checking if contract is a proxy. 
If this PR is merged I'll make a issue to indicate this need to be fixed at some point. I'd rather not add FIXME as it will probably cause our checks to fail.